ticketguardian/auth/auth.py

```diff
@@ -5,15 +5,38 @@ class Auth(RetrieveResourceMixin):
 
     resource = 'auth'
 
-    def get_scope(self):
+    @property
+    def scope(self):
         """
-        Get own scope.
-
+        Get scope of self.
         Returns:
             list -- A list containing all objects within scope including self
         """
         return self.retrieve('scope', raw_data=True, instance=self)
 
+    @classmethod
+    def get_parents(cls, affiliate_id):
+        """
+        Get all parents of an affiliate.
+
+        Returns:
+            list -- A list containing all parent ids of the Affiliate.
+        """
+        affiliate = Affiliate.retrieve(affiliate_id)
+
+        result = []
+
+        while hasattr(affiliate.parent, 'id'):
+            try:
+                result.append(affiliate.parent.id)
+                affiliate = affiliate.parent
+            except Exception:
```

Comment:
Do we know which exceptions we're expecting? If so, may be better to specify them:
`except (AttributeError, TypeError):`

Reply:
@alikashani Sadly the exception thrown here is a generic exception with a message returned from the API so in this case it would be `Exception: Not found.`